pkg/catalog/service.go

```diff
@@ -146,19 +146,21 @@ func (mc *MeshCatalog) GetTargetPortToProtocolMappingForService(svc service.Mesh
 func (mc *MeshCatalog) GetPortToProtocolMappingForService(svc service.MeshService) (map[uint32]string, error) {
 	portToProtocolMap := make(map[uint32]string)
 
-	k8sSvc := mc.kubeController.GetService(svc)
-	if k8sSvc == nil {
-		return nil, errors.Wrapf(ErrServiceNotFound, "Error retrieving k8s service %s", svc)
+	for _, provider := range mc.serviceProviders {
+		currentPortToProtocolMap, err := provider.GetTargetPortToProtocolMappingForService(svc)
+		if err != nil {
+			return nil, err
+		}
+		for key, value := range currentPortToProtocolMap {
+			if v, ok := portToProtocolMap[key]; ok && v != value {
+				return nil, errors.Errorf("Error fetching port:protocol, multiple entries found on same port for service %s", svc)
+			}
+			portToProtocolMap[key] = value
+		}
 	}
 
-	for _, portSpec := range k8sSvc.Spec.Ports {
-		var appProtocol string
-		if portSpec.AppProtocol != nil {
-			appProtocol = *portSpec.AppProtocol
-		} else {
-			appProtocol = kubernetes.GetAppProtocolFromPortName(portSpec.Name)
-		}
-		portToProtocolMap[uint32(portSpec.Port)] = appProtocol
```

Comment:
Why are these checks being removed?

Reply:
We moved  this check to the kube provider [pkg/providers/kube/client.go#L277](https://github.com/openservicemesh/osm/blob/bf225542d9e6569d7d6c638ab779be8be272c86c/pkg/providers/kube/client.go#L277) since the catalog shouldn't necessarily  know about kubernetes. @steeling